### ISSUE TITLE
problems in dist_train (single nodes multi-gpu)

### ISSUE THREAD

**Issue by Wyman-yang** [open-mmlab/mmdetection#3973]
I can run on single gpu with 0.02/8 learning rate, but I have some problems in distributed traing with two gpus on single node(lauhcher is pytorch).  I wirite a print function in model and I find it will stop after some iteration, and there will be no more log output, just like this:
![1](https://user-images.githubusercontent.com/61415784/96669303-86568400-138f-11eb-85ef-2725678f8424.JPG)
The numer I print is this:
![2](https://user-images.githubusercontent.com/61415784/96669370-ae45e780-138f-11eb-997e-fa8ba6f9dbfb.JPG)
The sbtach script is this:

- ![3](https://user-images.githubusercontent.com/61415784/96669474-dfbeb300-138f-11eb-9ead-967db0fc47bd.JPG)

The log didn't update just like this
![4](https://user-images.githubusercontent.com/61415784/96669587-15639c00-1390-11eb-924b-4b2162ca8df8.JPG)

I really need your help!! Thank you so much!

**Comment by Wyman-yang** [comment]
There is no error and the cmd"squeue" shows it is still running...

**Comment by Wyman-yang** [comment]
the version of my mmdet is 2.4.0 and mmcv-full is 1.1.3, all the codes on single gpu can work, but the pytorch launcher always cannot work.

**Comment by Wyman-yang** [comment]
> There is no error and the cmd"squeue" shows it is still running...

![5](https://user-images.githubusercontent.com/61415784/96671837-6033e280-1395-11eb-9460-3ac15d2dae11.JPG)
If I set inter 1 to print log, it will stuck after some iterations....

**Comment by RyanXLi** [comment]
Hi! Please refer to [this issue](https://github.com/open-mmlab/mmdetection/issues/3823) to see whether this is the problem.

**Comment by Wyman-yang** [comment]
> Hi! Please refer to [this issue](https://github.com/open-mmlab/mmdetection/issues/3823) to see whether this is the problem.
Thank you for your repy.

:( It still doesn't work. This has made me disappointed several days... 

I can train with single gpu with launcher "none". However, if I use pytorch launcher, the training iwill be stuck after some iterations. The log is as follows and there is no checkpoint saved and no new log printed after that.


![5](https://user-images.githubusercontent.com/61415784/96675651-f3711600-139d-11eb-851a-6aa77b48feb4.JPG)

**Comment by RyanXLi** [comment]
Actually, since you're using slurm you can try [slurm_train.sh](https://github.com/open-mmlab/mmdetection/blob/master/tools/slurm_train.sh) and see if the problem persists.

**Comment by Kyfafyd** [comment]
@tmp12316 
hi, could you please tell me how you finally solve this issue?
